java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -504,38 +504,33 @@ static void readDiskRanges(FSDataInputStream file,
     private static final class Key implements Comparable<Key> {
       private final int capacity;
       private final long insertionGeneration;
+      private final int hash;
 
       Key(int capacity, long insertionGeneration) {
         this.capacity = capacity;
         this.insertionGeneration = insertionGeneration;
+        this.hash = new HashCodeBuilder().append(capacity).append(insertionGeneration)
```

Comment:
I found that the equals and hashCode methods of the Key are not used at all, because the TreeMap implementation logic only use of compareTo. It is redundant to put the hashCode calculation statement in the constructor, and in fact there is no need to override them.

Reply:
I also agree with leave equals and hashCode. Just putting the hashcode calculation in the constructor seems redundant in the current implementation. 

ByteBufferAllocatorPool needs to support multiple calls to getBuffer with the same length before the buffer is returned for reuse. I think there may not be a suitable data structure other than TreeMap.

Reply:
sorry, I didn't notice revert hash cache.